### PR TITLE
[IMP] website_sale: minor improvements/cleanups in js files

### PR DESCRIPTION
**Pull request by Feyensv**
Lay the foundation for incoming tasks and work in e-commerce by progressively splitting the js logic in separate files for the different pages/flows.

### CODE REVIEW

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -1499,7 +1499,7 @@ def checkout(self, **post):
             return 'ok'
         return request.render("website_sale.checkout", values)
 
-    @route('/shop/cart/update_address', type='http', auth='public', methods=['POST'], website=True)
+    @route('/shop/cart/update_address', type='json', auth='public', methods=['POST'], website=True)
```
Comment (AntoineVDV) [inline addons/website_sale/controllers/main.py:1502]:
```suggestion
    @route('/shop/cart/update_address', type='json', auth='public', website=True)
```

File: addons/website_sale/static/src/js/checkout.js [hunk 2]
```diff
@@ -11,6 +11,14 @@ publicWidget.registry.websiteSaleCheckout = publicWidget.Widget.extend({
         'click .js_edit_address': '_onClickEditAddress',
     },
 
+    /**
+     * @constructor
+     */
+    init: function () {
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/checkout.js:17]:
```suggestion
    init() {
```

File: addons/website_sale/static/src/js/checkout.js [hunk 3]
```diff
@@ -11,6 +11,14 @@ publicWidget.registry.websiteSaleCheckout = publicWidget.Widget.extend({
...
+        this._super.apply(this, arguments);
+        this.rpc = this.bindService("rpc");
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/checkout.js:19]: simple quotes

**Review by AntoineVDV — APPROVED** [review]

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**